nano/lib/numbers.cpp

```diff
@@ -564,6 +564,79 @@ bool nano::uint128_union::decode_dec (std::string const & text)
 	return error;
 }
 
+bool nano::uint128_union::decode_dec (std::string const & text, nano::uint128_t scale)
+{
+	bool error (text.size () > 40 || (!text.empty () && text.front () == '-'));
+	if (!error)
+	{
+		auto decimal_point = std::use_facet<std::numpunct<char>> (std::locale ()).decimal_point ();
```

Comment:
I think this will always use the C locale because locale's default ctor ends up using `locale::classic` (C). The tests assume "." so maybe this is intentional? (in which case . could be hardcoded)

Reply:
I'm not sure if we should suppor other locales later (as it can cause confusions). Hardcoding back dot for now